swagger/medco-connector.yml

```diff
@@ -277,6 +291,16 @@ responses:
         result:
           $ref: "#/definitions/exploreQueryResultElement"
 
+  statusResponse:
+    description: "Node status response."
+    schema:
+      type: "object"
+      properties:
+        statusOK:
```

Comment:
This can be removed, and just rely on the returned status code (if 200 = OK).

Reply:
I am not sure it is a good idea to link the status code of the API to the status of MedCo. If I receive a 200 status code, I know that the API call worked correctly, meaning that it correctly identified the status of MedCo, which however can be either OK or NOT OK.

Reply:
It's a pretty standard way of using HTTP for REST APIs, 5xx indicates server errors (500 being the generic one), status fail is a server error. Here it just looks like duplication of information. As long as it is documented through the swagger file it is all good.

Reply:
I guess it's easier to rely on the http status code to get the result of the underlying computation, but I think it's better to keep separate the protocol layer (with its status codes) and the business logic layer (with its computation results). I think it's cleaner. I found some discussions which go in this direction:
https://softwareengineering.stackexchange.com/questions/305250/should-i-use-http-status-codes-to-describe-application-level-events
https://stackoverflow.com/questions/27921537/returning-http-200-ok-with-error-within-response-body/27922900
5xx errors should be returned only when something on the server did not handle the request as expected: https://www.baeldung.com/rest-api-error-handling-best-practices
Also several widespread API's, like the Facebook Graph API, follow this approach, i.e. returning a 200 status code with possibly a business logic error message in the response body: https://blog.runscope.com/posts/6-common-api-errors

This is an interesting discussion, and when we reach an agreement on that I think it deserves to be included in the development guidelines. For this reason I would like to hear  opinions also from @JoaoAndreSa and @Pro7ech.